x/cosmwasmpool/model/pool.go

```diff
@@ -81,9 +81,9 @@ func (p Pool) SpotPrice(ctx sdk.Context, quoteAssetDenom string, baseAssetDenom
 	}
 	response, err := cosmwasmutils.Query[msg.SpotPriceQueryMsg, msg.SpotPriceQueryMsgResponse](ctx, p.WasmKeeper, p.ContractAddress, request)
 	if err != nil {
-		return osmomath.Dec{}, err
+		return osmomath.BigDec{}, err
 	}
-	return osmomath.MustNewDecFromStr(response.SpotPrice), nil
+	return osmomath.MustNewBigDecFromStr(response.SpotPrice), nil
```

Comment:
Reviewer, please pay closer attention to this. I think it is acceptable to convert this to BigDec from string directly instead of string -> Dec -> BigDec but would like some more eyes.

Reply:
This seems fine to me

Reply:
This seems okay conceptually but would be great to have tests for it to be sure

Reply:
Thanks. 

On another look, will make an issue to error as opposed to panic in the conversion. Will try to see if I can test this without having to modify the transmuter contract as well